t/157-global-var.t

```diff
@@ -0,0 +1,489 @@
+# vim:set ft= ts=4 sw=4 et fdm=marker:
+
+use Test::Nginx::Socket::Lua;
+
+log_level('debug');
+
+repeat_each(2);
+
+plan tests => repeat_each() * (blocks() * 3 + 14);
+
+our $HtmlDir = html_dir;
+
+$ENV{TEST_NGINX_HTML_DIR} ||= html_dir();
+
+no_long_string();
+
+sub read_file {
+    my $infile = shift;
+    open my $in, $infile
+        or die "cannot open $infile for reading: $!";
+    my $cert = do { local $/; <$in> };
+    close $in;
+    $cert;
+}
+
+our $TestCertificate = read_file("t/cert/test.crt");
+our $TestCertificateKey = read_file("t/cert/test.key");
+
+add_block_preprocessor(sub {
+    my $block = shift;
+
+    if (!defined $block->error_log) {
+        $block->set_value("no_error_log", "[error]");
+    }
+
+    if (!defined $block->request) {
+        $block->set_value("request", "GET /t");
+    }
+
+});
+
+run_tests();
+
+__DATA__
+
+=== TEST 1: set_by_lua
+--- config
+    location /t {
+        set_by_lua_block $res {
+            if not foo then
+                foo = 1
+            else
+                ngx.log(ngx.WARN, "old foo: ", foo)
+                foo = foo + 1
+            end
+            return foo
+        }
+        echo $res;
+    }
+--- response_body_like eval
+qr/[12]/
+--- grep_error_log eval: qr/(old foo: \d+|write to the lua global variable '\w+')/
+--- grep_error_log_out eval
+["write to the lua global variable 'foo'\n", "old foo: 1\n"]
```

Comment:
Should cover the log level (`warn`) and the Lua backtrace (is there a backtrace logged at all?).